resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/MountedHadoopConfigurationsProvider.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes
+
+import java.io.File
+
+import com.google.common.base.Charsets
+import com.google.common.io.Files
+import io.fabric8.kubernetes.api.model.{ConfigMap, ConfigMapBuilder, EnvVar, EnvVarBuilder, Volume, VolumeBuilder, VolumeMount, VolumeMountBuilder}
+import scala.collection.JavaConverters._
+
+import org.apache.spark.internal.Logging
+
+// Note that we use ConfigMaps here as opposed to uploading through the Driver
+// REST server after the REST server has started. The reason for this is that
+// the REST server may itself need the Hadoop configuration files to download
+// remote dependencies.
+private[spark] case class MountedHadoopConfiguration(
+  confDirVolume: Volume,
+  confDirVolumeMount: VolumeMount,
+  confDirEnv: EnvVar,
+  confMap: ConfigMap)
+
+private[spark] case class MountedHadoopConfigurations(
+  hadoopConfDir: MountedHadoopConfiguration,
+  yarnConfDir: MountedHadoopConfiguration)
+
+private[spark] class MountedHadoopConfigurationsProvider(kubernetesAppId: String)
+    extends Logging {
+
+  def get(): MountedHadoopConfigurations = {
+    MountedHadoopConfigurations(
+      getMountedHadoopConfiguration(
+        confEnvVar = "HADOOP_CONF_DIR",
+        configMapName = s"$kubernetesAppId-hadoopConfDir",
+        mountPath = "/etc/hadoop/hadoopConfDir"),
```

Comment:
Why not `/etc/hadoop/conf` ? I think it's the default hadoop conf dir for most hadoop distros.